ibis/expr/tests/test_newrels.py

```diff
@@ -1038,11 +1038,148 @@ def test_join_chain_gets_reused_and_continued_after_a_select():
     assert abc._finish().op() == expected
 
 
-def test_join_name_collisions():
+def test_self_join_extensive():
     a = ibis.table(name="a", schema={"a": "int64", "b": "string"})
 
     aa = a.join(a, [a.a == a.a])
+    aa_ = a.join(a, "a")
+    aa__ = a.join(a, [("a", "a")])
```

Comment:
Since there's a lot of the same letter floating around, it would really help readability IMO if you added a couple comments stating what each new `aaaaaa` variable is meant to test.

Reply:
These are just API variant of the same call. Anyway, I would rather defer the rewrite of tests to unblock the follow-up work.